#463298 [C#テキスト]12章演習問題

kurosu-being opened the pull request:
第12章 シリアル化、逆シリアル化

主にシリアル化、逆シリアル化について学習し、JSONファイルやXMLファイルについて学習しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-1を実装しました。
Practice12-1-2のファイルを作成しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-2にEmployeeクラスのコレクションのオブジェクトを追加しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-2を実装しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-2において、不要なusingを削除しました。
Practice12-1-3を実装しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-4のファイルを作成しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-4を実装しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-4において、コメントを追加しました。
Practice12-2-1を実装しました。
Practice12-2-2のファイルを作成しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-4において、コメントを追加しました。
Practice12-2-1において、コメントを追加しました。
Practice12-2-2において、Novelistクラスを追加しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-2-2において、Novelistクラスの不要なusingを削除しました。
Practice12-2-2において、Project.csの不要なusingを削除しました

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-2において、不要なusingを削除しました。
Practice12-1-3において、不要なusingを削除しました
Practice12-2-1において、Novelistクラスのコメントを追加しました。
Practice12-2-2において、Novelistクラスのコメントを追加しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-1において、コンストラクタのコメントを追加しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-1において、戻り値のコメントを追加しました。

odaira1119 reviewed (changes requested):
少しコメントしました。
レビューコメントが反映されていませんでした。すみません。
再度、コメントしますので、お待ちください。

odaira1119 commented on 章末問題12/Practice12-1-2/Program12-1-2.cs:
```diff
@@ -0,0 +1,23 @@
+                new Employee { ID = 953, Name = "K", HireDate = new DateTime(2023, 4, 1)  },
+                new Employee { ID = 903, Name = "L", HireDate = new DateTime(2023, 4, 2)  },
+                new Employee { ID = 999, Name = "M", HireDate = new DateTime(2023, 4, 3)  }
+            };
```
オブジェクト初期化子でのインスタンス初期化を複数行っています。
同じインスタンス初期化が複数あるなら、コンストラクタを用意したほうが便利かつミスがなくなりそうです。
コンストラクタを定義して、それを使ってインスタンス初期化するようにしてください。

kurosu-being replied:
ご指摘ありがとうございます。
コンストラクタを利用して初期化が行えるように実装しました。

odaira1119 commented on 章末問題12/Practice12-1-3/Program12-1-3.cs:
```diff
@@ -0,0 +1,21 @@
+
+            using (var wReader = XmlReader.Create("employees.xml")) {
+                var wSerializer = new DataContractSerializer(typeof(Employee[]));
+                Employee[] wEmployees = wSerializer.ReadObject(wReader) as Employee[];
```
右辺から型が明らかですので、暗黙的な型宣言にしてください。

kurosu-being replied:
ご指摘ありがとうございます。
明示的な型を使用する形に修正いたしました。

odaira1119 commented on 章末問題12/Practice12-1-4/Program12-1-4.cs:
```diff
@@ -0,0 +1,27 @@
+                new Employee { ID = 953, Name = "K", HireDate = new DateTime(2023, 4, 1)  },
+                new Employee { ID = 903, Name = "L", HireDate = new DateTime(2023, 4, 2)  },
+                new Employee { ID = 999, Name = "M", HireDate = new DateTime(2023, 4, 3)  }
+            };
```
コンストラクタを活用してください。

kurosu-being replied:
ご指摘ありがとうございます。
コンストラクタを実装し、修正いたしました。

odaira1119 commented on 章末問題12/Practice12-2-2/Program12-2-2.cs:
```diff
@@ -0,0 +1,58 @@
+        static string SerializeNovelistToJson(Novelist vNovelist) {
+            var wSettings = new JsonSerializerSettings {
+
+            };
```
設定がデフォルトになっています。
何も設定しないなら、変数として管理する必要はないかと思いますので、
変数で宣言せずに使うときにインスタンス生成するようにしてください。

kurosu-being replied:
ご指摘ありがとうございます。
変数で管理せず、直接インスタンスを生成してシリアライズするように変更いたしました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-1-1,12-1-2,Practice12-1-4,Practice12-2-2において、コンストラクタを実装しました。

kurosu-being committed:
#463298 [C#テキスト]12章演習問題

Practice12-2-1において、問題文の誤字を修正しました。
Practice12-2-2において、不要なメソッドを削除し直接インスタンスを作成するように修正しました。

kurosu-being commented:
@ohdaira-being 
レビュー頂きありがとうございました！
お待たせしておりました、ご確認ください。
